src/corehost/common/pal.windows.cpp

```diff
@@ -227,6 +237,48 @@ bool pal::get_default_installation_dir(pal::string_t* recv)
     }
 
     append_path(recv, _X("dotnet"));
+
+    return true;
+}
+
+bool pal::get_sdk_self_registered_dir(pal::string_t* recv)
+{
+    DWORD size = 0;
+    const HKEY hkey = HKEY_LOCAL_MACHINE;
+	// The registry search occurs in the 32-bit registry in all cases.
+#if not defined(RRF_SUBKEY_WOW6432KEY)
+    #define RRF_SUBKEY_WOW6432KEY  0x00020000
+#endif
+    const DWORD flags = RRF_RT_REG_SZ | RRF_SUBKEY_WOW6432KEY;
+
+    //TODO: Use the get_arch() method here; includes arm & arm64
+#if defined(_TARGET_AMD64_)
+    const pal::char_t* sub_key = _X("SOFTWARE\\dotnet\\Setup\\InstalledVersions\\x64\\sdk");
+#else
+    const pal::char_t* sub_key = _X("SOFTWARE\\dotnet\\Setup\\InstalledVersions\\x86\\sdk");
+#endif
+
+    pal::char_t* value = _X("InstallLocation");
+
+    // Determine the size of the buffer
+    LONG result = ::RegGetValueW(hkey, sub_key, value, flags, nullptr, nullptr, &size);
+    if (result != ERROR_SUCCESS || size == 0)
+    {
+        return false;
+    }
+
+    // Get the key's value
+    std::vector<wchar_t> buffer(size/sizeof(wchar_t));
+    result = ::RegGetValueW( hkey, sub_key, value, flags, nullptr, &buffer[0], &size);
+    if (result != ERROR_SUCCESS)
+    {
+        return false;
+    }
+
+    pal::char_t* text(buffer.data());
+    recv->clear();
+    recv->assign(text);
```

Comment:
Can we just call `recv->assign(buffer.data())` ? - I don't see a need for the `text` local variable.
Also - is it necessary to call `clear` - I would think that `assign` overwrites completely previous value.